236. Lowest Common Ancestor of a Binary Tree

```diff
@@ -0,0 +1,50 @@
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
+ * };
+ */
+class Solution {
+    public:
+        TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
+            //visit the tree, find out how to get to p and q from root, then find out the common route.
+            vector<int> proute;
+            vector<int> qroute;
+            inorder(root, proute, p);
+            inorder(root, qroute, q);
+            int minl = min(proute.size(), qroute.size());
+            for (int i = 0; i < minl; ++i) {
+                if (proute[i] != qroute[i]) {
+                    return root;
+                }else if (proute[i] == 0) {
+                    root = root->left;
+                }else{
+                    root = root->right;
+                }
+            }
+            return root;
+        }
+        int inorder(TreeNode* root, vector<int>& route, TreeNode* target) 
+        {
+            //0 means left, and 1 means right
```

Comment:
quick reminder: if you are using numbers as an indicator or const number (coefficient etc.), then basically 2 ways to do that:

-  Using const. For example: int const LEFT = 1; 
-  Using enum. You can google enum and learn how to use it.

The reason we prefer those implementations is that it is more readable in code. People will know what you are saying.

Reply:
OK, now I use enum.